Server-side/src/projects/project.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable, Delete } from '@nestjs/common';
+import { CreateProjectDto } from './dto/create-project.dto';
+import { ObjectId } from 'mongoose';
+import { InjectModel } from '@nestjs/mongoose';
+import { Request } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class ProjectService {
+  constructor(
+    @InjectModel('project') public projectModel,
+    private readonly jwtService: JwtService,
+  ) {}
+  async create(createProjectDto: CreateProjectDto, req: Request) {
+    // console.log(req.header);
+    const authHeader = req.headers['authorization'];
+    const token = authHeader.split(' ')[1];
+    const decoded = await this.jwtService.verify(token, {
+      secret: process.env.JWT_SECRET,
+    });
+    // console.log(decoded.id);
+    const userId = decoded.id;
```

Comment:
You need to move this code to the AuthGuard to handle token verification and user extraction there.